### ISSUE TITLE
TorchScript fails when single_bin_mask is None in rtdl_num_embeddings.py

### ISSUE THREAD

**Issue by artem-panin** [Yura52/tabular-dl-num-embeddings#27]
When saving the model with TorchScript, the following line fails:
`model_scripted = torch.jit.script(self.model._orig_mod)`

The traceback indicates:
```
Unsupported operation: attempted to use multidimensional indexing on a non-tensor type
File ".../rtdl_num_embeddings.py", line 673
self.single_bin_mask[..., None],
```

This happens because self.single_bin_mask can be None if no single-bin features exist:
```
self.register_buffer(
    'single_bin_mask', single_bin_mask if single_bin_mask.any() else None
)
```

TorchScript does not allow indexing on NoneType, causing _save_model to fail.

**Suggested fix**
Ensure that self.single_bin_mask is always a tensor (even if empty):
```
buf = single_bin_mask if single_bin_mask.any() else torch.empty(0, dtype=torch.bool)
self.register_buffer('single_bin_mask', buf)
```

In forward pass, check if mask is empty:
```
        if x.shape[-1] > 1:
            x = torch.cat(
                [
                    x[..., :1].clamp_max(1.0),
                    x[..., 1:-1].clamp(0.0, 1.0),
                    (
                        x[..., -1:].clamp_min(0.0)
                        if self.single_bin_mask.numel() == 0
                        else torch.where(
                            # For features with only one bin,
                            # the whole "piecewise-linear" encoding effectively behaves
                            # like mix-max scaling
                            # (assuming that the edges of the single bin
                            #  are the minimum and maximum feature values).
                            self.single_bin_mask[..., None],
                            x[..., -1:],
                            x[..., -1:].clamp_min(0.0),
                        )
                    ),
                ],
                dim=-1,
            )
```
This makes TorchScript happy and avoids NoneType indexing.

Environment:
- Python version: 3.12
- PyTorch version: 2.6.0
- RTDL version: 0.0.12

**Comment by Yura52** [comment]
Hi, thank you for a detailed report! My understanding is that `torch.compile` is recommended today over TorchScript and other alternatives, and if I remember correctly, we successfully used `torch.compile` with feature embeddings when working on [TabM](https://github.com/yandex-research/tabm). So I am a bit hesitant to optimize the implementation specifically for TorchScript :) By any chance, did you find a workaround for your use case?

P.S. Also, I am curious in what cases TorchScript is preferred today over `torch.compile`?

**Comment by artem-panin** [comment]
@Yura52 Hi, thanks for the response!

You can check the commit with the workaround in my fork:
https://github.com/artem-panin/rtdl-num-embeddings/commit/980b5d77b05479a03cd9b9261ea0980fe23be955

We use convertation to the TorchScript format for further deployment to production inference written on Go

**Comment by Yura52** [comment]
I see, thanks for sharing! Do I understand correctly that the issue is not a blocker, since you have the fork with a workaround? I am leaning to closing the issue to avoid TorchScript-specific tricks.

**Comment by artem-panin** [comment]
Yep, that's not a blocker for us
